src/librustc/hir/mod.rs

```diff
@@ -1450,7 +1450,7 @@ pub struct ItemId {
 /// The name might be a dummy name in case of anonymous items
 #[derive(Clone, PartialEq, Eq, RustcEncodable, RustcDecodable, Hash, Debug)]
 pub struct Item {
-    pub name: Name,
+    pub name: Spanned<Name>,
```

Comment:
You don't even need to go through all this. `recursive_type_with_infinite_size_error` is only called in `librustc_typeck/check/mod.rs`, and it already contains an `item_id` that you can use.

So instead of passing in a `DefId`, pass in an `ast::NodeId`, and then use `self.map.expect_item(item_id)`, and you'll get the ast::Item.